R/tools.R

```diff
@@ -37,17 +83,25 @@ lint_dir <- function(path) {
 #' with the `legacy_max_lint_r_errors` option in `rhino.yml`.
 #' This can be useful when inheriting legacy code with multiple styling issues.
 #'
+#' @param paths Character vector of directories and files to lint.
+#' When `NULL` (the default), check `app` and `tests/testthat` directories.
+#'
 #' @return None. This function is called for side effects.
 #'
 #' @export
-lint_r <- function() {
+lint_r <- function(paths = NULL) {
+  if (is.null(paths)) {
+    paths <- c("app", "tests/testthat")
+  }
+
+  paths <- check_paths(paths = paths)
+
   max_errors <- read_config()$legacy_max_lint_r_errors
+
   if (is.null(max_errors)) max_errors <- 0
```

Comment:
Opinionated nitpicks:
1. Blank lines can make the code less readable. This note on [vertical whitespace](https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace) gives some remarks which I very much agree with (it's from Google's C++ style guide, but I think it applies to any language).
2. When the argument's meaning is obvious (as is typically the case with single-argument functions), it's better to omit its name (for readability and ease of refactoring).

```suggestion
  if (is.null(paths)) {
    paths <- c("app", "tests/testthat")
  }
  paths <- check_paths(paths)

  max_errors <- read_config()$legacy_max_lint_r_errors
  if (is.null(max_errors)) max_errors <- 0
```